pyspectrafuse/common/duckdb_ops.py

```
@@ -0,0 +1,531 @@
+    Args:
+        window_tsvs: List of MaRaCluster output TSV paths, one per window.
+        window_mz_ranges: List of (min_mz, max_mz) tuples for each window.
+        overlap_da: Overlap width in Da.
+
+    Returns:
+        DataFrame with columns: mgf_path, scannr, cluster_id (resolved).
+    """
+    conn = _create_connection()
+    try:
+        all_parts = []
+        for i, (tsv_path, (min_mz, max_mz)) in enumerate(zip(window_tsvs, window_mz_ranges)):
```

Ruff (0.15.10)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)